packages/vite/src/node/server/middlewares/proxy.ts

```diff
@@ -53,6 +53,7 @@ export function proxyMiddleware(
 
     if (opts.configure) {
       opts.configure(proxy, opts)
+      delete opts.configure
```

Comment:
Is there a drawback if we don't `delete` this? Hope you can help me understand this.

Reply:
Its basically nitpick memory consumption reduction. Its applied then passed to http-proxy which has no use of it as an option.

Reply:
I'm still a bit wary of this change though. `delete opts.configure` would actually mutate the vite config if you follow the references before the code here. Perhaps the fix should be done [here](https://github.com/vitejs/vite/pull/7361/files#diff-b46b4f407fe94435eeb7d1c687d762435fb90d12208a4a1eb7b4883b947b29c2R59). Something like:

```js
const { configure, others, ...proxyOpts } = opts
```

But I do wonder how much gain we get from this.

Reply:
> But I do wonder how much gain we get from this.

Very little but still something. If you want me to close the PR, all good.

Reply:
I think we can close this for now until there's an issue that arises from it :)